### PR TITLE
refactor: Upgrade to mime 4.0.4

### PR DESCRIPTION
**Pull request by Moumouls**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: https://github.com/parse-community/parse-server/pull/9248

## Tasks

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)
- [x] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [x] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

File: spec/FilesController.spec.js [hunk 1]
```diff
@@ -11,8 +11,8 @@ const mockAdapter = {
   createFile: () => {
     return Promise.reject(new Error('it failed with xyz'));
   },
-  deleteFile: () => {},
-  getFileData: () => {},
+  deleteFile: () => { },
```
Comment (mtrezza) [inline spec/FilesController.spec.js]: There seems to be something going on with your IDE doing this refactoring. Could you revert these? Maybe some lint or prettier config we need to change in the repo.

File: src/Routers/FilesRouter.js [hunk 2]
```diff
@@ -67,7 +66,7 @@ export class FilesRouter {
     return router;
   }
 
-  getHandler(req, res) {
+  async getHandler(req, res) {
```
Comment (mtrezza) [inline src/Routers/FilesRouter.js:69]: Can this just be changed to async without any other changes in the calling code?

Comment (Moumouls) [reply src/Routers/FilesRouter.js:69]: yes the getHandler is just used in an express route, and express don't care about async or sync callbacks

File: spec/FilesController.spec.js (hunk 1 above)
Comment (Moumouls) [reply spec/FilesController.spec.js]: i'll check

Comment (Moumouls) [reply spec/FilesController.spec.js]: @mtrezza  my IDE seems right, and use the provided .prettierrc on save.

It seems that parse-server have a lot of non formatted files. I just ran the prettier command from the package.json and it result in many changes

image

parse-server don't have pre commit system to ensure that each touched file follow prettier rules.

May be something to add

Comment (mtrezza) [reply spec/FilesController.spec.js]: Yes, that's an open issue we need to address. We don't run prettier currently, and before we run it across the whole repo, we'd need to review the prettier settings. So thats a separate issue. I've reverted the unrelated changes and could you please ignore prettier for now in other PRs?

**Comment by Moumouls** [comment]
ready to merge @mtrezza

**Comment by Moumouls** [comment]
@mtrezza are you aware of a possible flaky on Mongo 8 CI, since this PR don't touch to mongo it self and there is some failure ?

File: spec/FilesController.spec.js [hunk 3]
```diff
@@ -11,8 +11,8 @@ const mockAdapter = {
...
+  getFileData: () => { },
```
Comment (mtrezza) [inline spec/FilesController.spec.js]:
```suggestion
  getFileData: () => {},
```

File: spec/FilesController.spec.js (hunk 1 above)
Comment (mtrezza) [inline spec/FilesController.spec.js]:
```suggestion
  deleteFile: () => {},
```

**Comment by mtrezza** [comment]
> @mtrezza are you aware of a possible flaky on Mongo 8 CI, since this PR don't touch to mongo it self and there is some failure ?

We can always re-run the CI to see; is this ready for merge, once the CI passes?

**Comment by Moumouls** [comment]
yes @mtrezza  ready for merge

**Review by mtrezza — APPROVED** [review]

**Pull request merged**